python/sqlflow_submitter/xgboost/dataset.py

```diff
@@ -135,8 +140,12 @@ def thread_worker(slice_id):
             ]))
         complete_queue.put(slice_id)
 
-    for i in range(SLICE_NUM):
-        t = threading.Thread(target=thread_worker, args=(i, ))
+    slice_id = rank
+    slice_total = 0
+    while slice_id < SLICE_NUM:
```

Comment:
It seems there is an assumption that `SLICE_NUM` always greater than number of workers. Should check this some how.

Reply:
I'm going to add a checker for `train.num_workers`: `attribute.IntRangeChecker(1, 128, true, true)}`

However, I prefer to adjust `SLICE_NUM` by `num_workers`. Let me add a tag `TODO` here.